src/templates/module_test.html

```diff
@@ -0,0 +1,26 @@
+<!-- Bootstrap HTML for running the unit tests. -->
+<!DOCTYPE html>
+<html>
+  <head>
+    <script type="text/javascript">
+      window.logs = [];
+      console.log = function (message) {
+        window.logs.push(message);
+      };
+      console.warn = function (message) {
+        window.logs.push(message);
+      };
+      console.info = function (message) {
+        window.logs.push(message);
+      };
+      console.error = function (message) {
+        window.logs.push(message);
+      };
+    </script>
+    <script type="module">
+      import { loadPyodide } from "./pyodide.mjs";
+      window.loadPyodide = loadPyodidea;
```

Comment:
I guess this must be a typo?

Reply:
huh I wonder why that wouldn't break the tests

Reply:
Hmm, that's interesting. Let me double check ~

Reply:
Okay I think the tests pass because we already attach `loadPyodide` to [`globalThis` here](https://github.com/pyodide/pyodide/blob/main/src/js/pyodide.ts#L271), while this typoed line errors out due to undefined 
 `loadPyodidea`, so the line is not doing anything.

Reply:
Thanks for the investigation. In that case we don't need to bind `loadPyodide` to the global in the test.html file anymore, so we would better remove those.

Reply:
Well ideally we should eventually update the module build to avoid the global entrypoint. The `(globalThis as any).loadPyodide = loadPyodide;` is really only there for the non-ES6-module build. What we should do is have an extra file which just does
```js
import {loadPyodide} from "./pyodide.ts";
(globalThis as any).loadPyodide = loadPyodide;
```
and then build `pyodide.js` from this file and build `pyodide.mjs` from `pyodide.ts`.

Reply:
So I'm guessing we may not necessarily need the `globalThis` line in `pyodide.ts`, since rollup will automatically attach all exports to a global variable specified in [the rollup config](https://github.com/pyodide/pyodide/blob/main/src/js/rollup.config.js#L13). I made a very simple example on umd vs esm build with rollup [here](https://stackblitz.com/edit/node-51syxy?file=bundle.js). In current implementation, we have `loadPyodide`  as a global variable. If we remove [this line](https://github.com/pyodide/pyodide/blob/main/src/js/pyodide.ts#L271). We should still be able to access the function in the browser via `loadPyodide.loadPyodide`, though it's probably better to rename the rollup config name as pyodide so we can access via `pyodide.loadPyodide`. But that also means we are changing the API..

Reply:
@hoodmane Sorry  just saw your comments. Yeah I think even for non es6 build, it's ok to remove the line and let rollup handle the global variable binding for us. But that also means we are changing what pyodide exposes as global variable.

Reply:
@ryanking13 Yeah, if we keep current implementation, then there's no need for that line in `test.html` file.

Reply:
Yeah I don't want to make an API change like that. Currently we expose a function `loadPyodide` which returns `pyodide`. This makes sense to me. `pyodide.loadPyodide` or `loadPyodide.loadPyodide` are not so great. But I think we can easily just use a different entrypoint for the non-module version.

Incidentally, what I would really like is to bundle all of our top level code into a function called `loadPyodide` and expose that. We don't need any global variables currently, so there is nothing preventing loading two copies of Pyodide next to each other. But currently we use a lot of top level variables and I don't understand how to restructure the logic to avoid that while still allowing `import { func} from "./file"` and the doc generation stuff etc to work. I don't *think* rollup has "bundle everything into one big function" as an option.

Reply:
I'm not sure what kind of top level variables / doc generation you are referring to. But to clarify, rollup bundle everything into an object and exposes that as a global variable, if you check the minimum example I linked above: https://stackblitz.com/edit/node-51syxy?file=bundle.js. which basically simplifies to something like the following:

```
(function (global, factory) {
  factory(global.pyodide = {});
}(this, (function (exports) { 'use strict';
  function loadPyodide() {
    console.log('loading pyodide');
  }
  exports.loadPyodide = loadPyodide;
  Object.defineProperty(exports, '__esModule', { value: true });
})));
```

Reply:
> I'm not sure what kind of top level variables / doc generation you are referring to.

Well for instance in `module.ts` we define `export let Module: any = {};` which gets the emscripten module loaded into it. So you can't use `loadPyodide` twice because they would end up sharing the same emscripten module.

Reply:
So ideally I would want rollup to create a (non es6 module) bundle that looks something like:
```js
function loadPyodide(config) {
  let module_exports;
  function init_module() {
    if (module_exports) {
      return;
    }
    module_exports = {};
    // top level contents of module.ts
    let Module = {};
    module_exports.Module = Module;
  }
  let load_pyodide_exports;
  function init_load_pyodide() {
    if (load_pyodide_exports) {
      return;
    }
    load_pyodide_exports = {};
    init_module();
    let Module = module_exports.Module;
    //... top level contents of load-pyodide.ts
  }
  // ...
  let pyodide_exports;
  function init_pyodide() {
    if (pyodide_exports) {
      return;
    }
    pyodide_exports = {};
    init_load_pyodide();
    let initializePackageIndex = load_pyodide_exports.initializePackageIndex;
    let loadPackage = load_pyodide_exports.loadPackage;
    // ... top level contents of pyodide.ts
    function loadPyodideInner(config) {
      // ...
    }
    pyodide_exports.loadPyodideInner = loadPyodideInner;
  }
  init_pyodide();
  pyodide_exports.loadPyodideInner(config);
}
window.loadPyodide = loadPyodide
```

Reply:
For now I opened #2249 as a followup to this.

Reply:
I'm not sure I understand. But isn't this what it's doing now ? The exports in `module.ts` are not exposed since it's not the entry point to rollup. Only exports in the entry point will be exposed publicly.

Reply:
Oh I think I understand what you mean, when we call `loadPyodide` multiple times, we are overwriting the same Module instance.

Reply:
Yeah exactly.

Reply:
I personally don't have much use case to load multiple pyodide instances yet. But I think it could definitely be something useful. Maybe we could wrap modules and `loadPyodide` in a class `Pyodide` and expose that class. Then every time we instantiate a new instance of `Pyodide`, we create a  new standalone module that does not interfere with other instances.

Reply:
Yeah maybe that would work.

Reply:
See #2255.